src/main/kotlin/com/terraformation/backend/tracking/db/ObservationStore.kt

```diff
@@ -1162,118 +1164,248 @@ class ObservationStore(
     }
 
     withLockedObservation(observationId) { observation ->
-      val observationPlotDetails =
-          dslContext
-              .select(
-                  OBSERVATION_PLOTS.MONITORING_PLOT_ID,
-                  OBSERVATION_PLOTS.IS_PERMANENT,
-                  MONITORING_PLOTS.PLANTING_SUBZONE_ID,
-                  PLANTING_SUBZONES.PLANTING_ZONE_ID)
-              .from(OBSERVATION_PLOTS)
-              .join(MONITORING_PLOTS)
-              .on(OBSERVATION_PLOTS.MONITORING_PLOT_ID.eq(MONITORING_PLOTS.ID))
-              .leftJoin(PLANTING_SUBZONES)
-              .on(MONITORING_PLOTS.PLANTING_SUBZONE_ID.eq(PLANTING_SUBZONES.ID))
-              .where(OBSERVATION_PLOTS.OBSERVATION_ID.eq(observationId))
-              .fetch()
-              .associateBy { it[OBSERVATION_PLOTS.MONITORING_PLOT_ID]!! }
+      when (observation.observationType) {
+        ObservationType.BiomassMeasurements ->
+            mergeOtherSpeciesForBiomass(observation, speciesId, otherSpeciesName)
+        ObservationType.Monitoring ->
+            mergeOtherSpeciesForMonitoring(observation, speciesId, otherSpeciesName)
+      }
+    }
+  }
+
+  private fun mergeOtherSpeciesForBiomass(
+      observation: ExistingObservationModel,
+      speciesId: SpeciesId,
+      otherSpeciesName: String
+  ) {
+    val observationId = observation.id
+
+    val otherBiomassSpeciesId =
+        with(OBSERVATION_BIOMASS_SPECIES) {
+          dslContext.fetchValue(
+              ID, SCIENTIFIC_NAME.eq(otherSpeciesName).and(OBSERVATION_ID.eq(observationId)))
+        }
+
+    val targetBiomassSpeciesId =
+        with(OBSERVATION_BIOMASS_SPECIES) {
+          dslContext.fetchValue(ID, SPECIES_ID.eq(speciesId).and(OBSERVATION_ID.eq(observationId)))
+        }
+
+    if (otherBiomassSpeciesId == null) {
+      log.warn(
+          "Biomass observation $observationId does not contain species name $otherSpeciesName; " +
+              "merge is a no-op")
+      return
+    }
```

Comment:
Technically this fetch could be moved to below the check for `otherBiomassSpeciesId`, to avoid an unnecessary fetch if other is null

Reply:
Ha, I had it that way originally, then thought, "It looks nicer to have the two `if`s next to each other."

Reply:
Ha! I'm good either way. I should have added `NIT` to the front of my comment. I see you already made the change so I'll leave it up to you. The PR is approved either way